app/views/github_oauth.py

```diff
@@ -1,71 +1,47 @@
 import datetime
-import os
 
-from dotenv import load_dotenv
 from rest_framework import status
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
 from rest_framework_simplejwt.tokens import RefreshToken
 
-from ..oauth import (convert_to_auth_token, generate_github_access_token,
-                     get_user_from_token, retrieve_github_user_info)
-
+from ..github_oauth import generate_github_access_token, retrieve_github_user_info
+from ..models import User
 from ..serializers import UserSerializer
 
-load_dotenv()
-
-# GITHUB ID AND SECRET
-SOCIAL_AUTH_GITHUB_KEY = os.getenv('SOCIAL_AUTH_GITHUB_KEY')
-SOCIAL_AUTH_GITHUB_SECRET = os.getenv('SOCIAL_AUTH_GITHUB_SECRET')
-
-# OAUTH TOOLKIT ID AND SECRET
-CLIENT_ID = os.getenv('CLIENT_ID')
-CLIENT_SECRET = os.getenv('CLIENT_SECRET')
-
 
 def get_refresh_access_token(user_object):
     refresh = RefreshToken.for_user(user_object)
     access_token = refresh.access_token
     access_token.set_exp(lifetime=datetime.timedelta(minutes=480))
-    return {
-        "refresh": str(refresh),
-        "access": str(access_token)
-    }
+    return {"refresh": str(refresh), "access": str(access_token)}
 
 
-@api_view(['POST'])
+@api_view(["POST"])
 def github_authenticate(request):
     github_token = generate_github_access_token(
-        github_client_id=SOCIAL_AUTH_GITHUB_KEY,
-        github_client_secret=SOCIAL_AUTH_GITHUB_SECRET,
-        github_code=request.data['code']
+        github_code=request.data["code"],
     )
 
-    github_user = retrieve_github_user_info(
-        token=github_token
-    )
+    github_user = retrieve_github_user_info(token=github_token)
 
-    django_auth_token = convert_to_auth_token(
-        client_id=CLIENT_ID,
-        client_secret=CLIENT_SECRET,
-        backend='github',
-        token=github_token
+    user, created = User.objects.get_or_create(
+        username=github_user.get("name"),
+        avatar=github_user.get("avatar_url"),
+        email=github_user.get("email"),
+        timezone=github_user.get("location"),
     )
 
-    user = get_user_from_token(django_auth_token)
-
-    user.avatar = github_user.get('avatar_url')
-    user.timezone = github_user.get('location')
-    user.save()
+    if user is None and created is False:
+        print("user not created")
```

Comment:
You need to stop the flow execution here, because just printing alone will not cause this function flow to stop. You might want to return a response with the appropriate status code for this kind of scenario.